mlflow/models/evaluation/base.py

```diff
@@ -151,6 +156,170 @@ def __str__(self):
+
+    Args:
+        eval_fn: the evaluation function of the EvaluationMetric.
+        require_strict_signature: (Optional) Whether the eval_fn needs to follow a strict signature.
+            If True, then the eval_fn must follow below signature:
+
+                .. code-block:: python
+
+                    def eval_fn(
+                        predictions: "pd.Series",
+                        metrics: Dict[str, MetricValue],
+                        inputs: "pd.Series",
+                        *args,
+                    ) -> MetricValue:
+                        pass
+
+            When generating a metric from `make_genai_metric`, this should be set to True.
+            Default to False.
+
+    Returns:
+        A dynamically generated callable CallableEvaluationMetric class.
+    """
+    from mlflow.metrics.base import MetricValue
+
+    if require_strict_signature:
+        allowed_kwargs_names = [
+            param_name
+            for param_name in inspect.signature(eval_fn).parameters.keys()
+            if param_name not in ["predictions", "metrics", "inputs"]
+        ]
+
+        def genai_call_method(
+            self,
+            *,
+            predictions: Union[pd.Series, str, list],
+            inputs: Union[pd.Series, str, list],
+            metrics: Optional[Dict[str, MetricValue]] = None,
+            **kwargs,
+        ) -> Union[MetricValue, List[str], List[float]]:
+            if missed_kwargs := set(allowed_kwargs_names) - set(kwargs.keys()):
+                raise MlflowException.invalid_parameter_value(
+                    f"Missing required arguments: {missed_kwargs}",
+                )
+            if extra_kwargs := set(kwargs.keys()) - set(allowed_kwargs_names):
+                raise MlflowException.invalid_parameter_value(
+                    f"Unexpected arguments: {extra_kwargs}",
+                )
+            return self.eval_fn(
+                _convert_val_to_pd_Series(predictions, "predictions"),
+                metrics or {},
+                _convert_val_to_pd_Series(inputs, "inputs"),
+                # Note: based on https://github.com/mlflow/mlflow/blob/4fef77afdbe4d76302cb0b1aad2bd72b5cde64e9/mlflow/metrics/genai/genai_metric.py#L49-L53
+                # the extra params passed https://github.com/mlflow/mlflow/blob/4fef77afdbe4d76302cb0b1aad2bd72b5cde64e9/mlflow/metrics/genai/genai_metric.py#L513
+                # should always be pandas Series
+                *[
+                    _convert_val_to_pd_Series(kwargs[arg_name], arg_name)
+                    for arg_name in allowed_kwargs_names
+                ],
+            )
+
+        genai_call_method.__signature__ = Signature(
+            parameters=[
+                Parameter("self", Parameter.POSITIONAL_OR_KEYWORD),
+                Parameter(
+                    "predictions",
+                    Parameter.KEYWORD_ONLY,
+                    annotation=Union[pd.Series, str, list],
```

Comment:
```suggestion
                    annotation=Union[pd.Series, str, List[???]],
```

what's the element type? str? Any?

Reply:
Looked through the internal implementation, we use prompt template to format input, output, so this should be List[str]!